### ISSUE TITLE
Support for Cassandra 3.x

### ISSUE THREAD

**Issue by mikesparr** [Mashape/kong#831]
Using the latest Kong 0.5.4 and connecting to a 4-node Cassandra cluster fails on startup. In reading on the error it implies that the database client driver is not up-to-date to support latest version(s) of db. Is there support for Cassandra 3.x or how can this be resolved?

Console with error upon startup
~~~
ubuntu@app1:~$ sudo kong start
[INFO] Using configuration: /etc/kong/kong.yml
[INFO] Kong version.......0.5.4
       Proxy HTTP port....8000
       Proxy HTTPS port...8443
       Admin API port.....8001
       dnsmasq port.......8053
       Database...........cassandra data_centers= strategy_class=NetworkTopologyStrategy timeout=1000 contact_points=db1:9042,db2:9042,db3:9042,db4:9042 keepalive=60000 keyspace=kong
[INFO] Connecting to the database...
[ERR] Cassandra error: Failed to prepare statement: "SELECT * FROM system.schema_keyspaces WHERE keyspace_name = ?;". Cassandra returned error (Invalid): unconfigured table schema_keyspaces
~~~

Cassandra versions:
~~~
ubuntu@db2:~$ cqlsh db2
Connected to my-cluster-name at db2:9042.
[cqlsh 5.0.1 | Cassandra 3.1.1 | CQL spec 3.3.1 | Native protocol v4]
Use HELP for help.
~~~

Kong config (database section):
~~~
## Specify the DAO to use
database: cassandra

## Databases configuration
databases_available:
  cassandra:
    properties:
      contact_points:
        - "db1:9042"
        - "db2:9042"
        - "db3:9042"
        - "db4:9042"
      timeout: 1000
      keyspace: kong
      keepalive: 60000 # in milliseconds
      strategy_class: NetworkTopologyStrategy
      data_centers:
        DC1: 2
      # ssl: false
      # ssl_verify: false
      # ssl_certificate: "/path/to/cluster-ca-certificate.pem"
      # user: cassandra
      # password: cassandra
~~~

Any tips are appreciated. I moved from local db instance on app server to centralized cluster so installed latest db versions and now it fails. I don't want to downgrade db to 2.x versions.

**Comment by thibaultCha** [comment]
The driver needs to be tested against 3.0 and eventually updated to support it. Ideally it would also have to support the binary protocol v4. Then, Kong needs to be tested and eventually updated to support 3.0 too, if any change is necessary. 

Since Cassandra 2 is currently considered the most stable version, we have not considered supporting 3.0 because of other priorities. In the future, we might still consider supporting other databases such as Postgres a higher priority than supporting Cassandra 3.0.

**Comment by mikesparr** [comment]
Thanks for quick reply. It looks like I can downgrade and pin `2.2.4` for now until `3.x` support is available. Serves me right for wanting the latest and greatest but realized our local dev is on `2.2.3` anyway so will be in sync. ;-)

**Comment by mikesparr** [comment]
Ansible playbook tweak to deb version and pins and re-run... we are back in business in 20 minutes. Thanks again and whenever 3.x supported we'll upgrade here too.

**Comment by SandeshSarfare** [comment]
I cant see kong config anywhere

**Comment by thibaultcha** [comment]
#1709 has been merged! This will land in 0.10 scheduled by the end of the month.

**Comment by jaylinwang** [comment]
😈 i have a same problem, but find this issue now. expect for 0.10 publish
